scripts/longtermeffects.lua

```diff
@@ -28,6 +28,111 @@ function onClose()
 	CombatManager2.clearExpiringEffects = clearExpiringEffects_old;
 end
 
+local function filterTable (table, filterFunction)
+	local tFiltered = {}
+	for key, value in pairs(table) do
+		if filterFunction(value) then
+			table.insert(tFiltered, key, value)
+		end
+	end
+	return tFiltered
+end
+
+local function inTable (table, checkedValue)
+	for _, value in pairs(table) do
+		if value == checkedValue then
+			return true
+		end
+	end
+	return false
+end
+
+local function splitEffectIntoComponentsTypes(sEffect)
+	local aEffectComps = EffectManager.parseEffect(sEffect)
+	local aComponentTypes = {}
+	for index, effectComp in ipairs(aEffectComps) do
+		local component = EffectManager.parseEffectCompSimple(effectComp).type
+		table.insert(aComponentTypes, index, component)
+	end
+	return aComponentTypes
+end
+
+local function EffectTypeShouldBeChecked(sEffectComponentType)
+	local arrsComponentsToInclude = {'FHEAL', 'REGEN', 'DMGO'}
+	return inTable(arrsComponentsToInclude, sEffectComponentType)
+end
+
+local function ActorRequiresSlowMode(actor, arrSEffects)
+
+	local actorHealth = ActorHealthManager.getHealthStatus(actor)
+
+	-- Has ongoing damage, and still lives.
+	if inTable(arrSEffects, 'DMGO') then
+		if actorHealth ~= ActorHealthManager.STATUS_DEAD then
+			return true
+		end
+	end
+
+	-- Healing through Regeneration
+	if inTable(arrSEffects, 'REGEN') then
+		if actorHealth ~= ActorHealthManager.STATUS_HEALTHY then
+			return true
+		end
+	end
+	-- Healing through Fast Healing
+	if inTable(arrSEffects, 'FHEAL') then
+		if actorHealth ~= ActorHealthManager.STATUS_HEALTHY and actorHealth ~= ActorHealthManager.STATUS_DEAD then
+			return true
+		end
+	end
+	return false
+end
+
+local function IsActorDying(actor, bIsStable)
+	local actorHealth = ActorHealthManager.getHealthStatus(actor)
+	if not bIsStable and actorHealth == ActorHealthManager.STATUS_DYING then
+		return true
+	end
+	return false
+end
+
+local function getIsStableAndEffectsToCheck(nodeCT)
+	-- Returns if node has effect stable, and flat list of all effect types.
+	-- Does two thing at once. as I dont want to iterate twice over all effects 
+	local aEffectsRequiringSlowMode = {}
+	local bIsCTStable = false
+	for _, nEffect in pairs(DB.getChildren(nodeCT, 'effects')) do
+		local sEffect = EffectManager.getEffectString(nEffect, false)
+		if string.lower(sEffect) == 'stable' then
+			bIsCTStable = true
+		else
+			local splitEffectComps = splitEffectIntoComponentsTypes(sEffect)
+			local splitSimulatedEffectComps = filterTable(splitEffectComps, EffectTypeShouldBeChecked)
```

Comment:
EffectTypeShouldBeChecked is a function call and needs the sEffectComponentType

Reply:
This is my first time working in lua, so I didn't know you could pass functions in function calls.